ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -1230,6 +1236,77 @@ object Serializers {
       }
     }
 
+    /** Patches the argument of a `Throw` for IR version until 1.8.
+     *
+     *  Prior to Scala.js 1.11, `Throw(e)` was emitted by the compiler with
+     *  the somewhat implied assumption that it would "throw an NPE" (but
+     *  subject to UB so not really) when `e` is a `null` `Throwable`.
+     *  In Scala.js 1.11, we explicitly ruled out this behavior, so that
+     *  `Throw(e)` only ever throws the value of `e`, while the NPE UB is
+     *  specified by `UnwrapFromThrowable`. Among other things, this allows the
+     *  user-space code `js.special.throw(e)` to indiscriminately throw `e`
+     *  even if it is `null`.
```

Comment:
I think we should add here that: Pre Scala.js 1.11, there was no direct user-space access to `Throw` and that as a result, the Scala.js compiler never generated a bare `Throw(null)` with the intended semantics of actually throwing null.

IIUC without this condition, the whole premise of the hack is moot.

Reply:
Alternatively: Maybe it is better if we simply clarify that Scala.js versions pre 1.11 will not work with checked NPEs for exceptions (and remove the hack entirely).

Reply:
> IIUC without this condition, the whole premise of the hack is moot.

Yes, indeed. Good point.

> Alternatively: Maybe it is better if we simply clarify that Scala.js versions pre 1.11 will not work with checked NPEs for exceptions (and remove the hack entirely).

I did contemplate this solution for a while. But since we can have a hack that makes it work always, IMO it's worth it. (It wouldn't be worth it if other unhackable aspects of the IR had to be fixed, but it's the only one.)